### ISSUE TITLE
[BUG] nix build test is flaky while building for MacOS 

### ISSUE THREAD

**Issue by vados-cosmonic** [wasmCloud/wash-archive#628]
**Describe the bug**
Periodically, when running the `nix` build for MacOS it fails (throwing the 137 for OOMKill, then 9 for SIGKILL):

```
copying path '/nix/store/vf9irc1alx9xvddff63asdrlhv1imm6g-gmp-with-cxx-6.2.1' from 'https://cache.nixos.org/'...
copying path '/nix/store/2n2i79dlpv559mwy7my7jf41agm6rf1s-coreutils-9.3' from 'https://cache.nixos.org/'...
/Users/runner/work/_temp/32bf7a0b-acce-4894-ba81-c3b380b56c34.sh: line 1: 30825 Killed: 9               nix run -L --inputs-from . 'nixpkgs#coreutils' -- --coreutils-prog=ginstall -p "./result/bin/wash" 'wash-x86_64-apple-darwin'
Error: Process completed with exit code 137.
```

The problem seems to be the use of the GNU `install` provided by [`nixpkgs#coreutils`  in `action.yml`](https://github.com/wasmCloud/wash/blob/main/.github/actions/build-nix/action.yml#L15) (the assumption here being that there is too much memory used by installing the coreutils install).

A possible solution is to replace that with platform specific solutions for installing the binary.

There don't seem to be many other solutions for `nix` memory usage discussed online other than enabling `swap`, but there is also the option of [using a MacOS XL runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software)

**Once this is finished**, `continue-on-error` should be removed from `.github/workflows/nix.yml`

**To Reproduce**
1. Run the `wash-x86_64-apple-darwin` CI build
2. Observe an unexplained failure (signal 137 -> signal 9)
3. See error

**Expected behavior**

The build should pass consistently

**Screenshots**

**Environment (please complete the following information) **
 - OS: MacOS
 - Shell: zsh
 - wash Version `next`

**Additional context**

**Comment by vados-cosmonic** [comment]
This has a chance of being [solved by using a larger runner](https://github.com/wasmCloud/wash/pull/619/files#diff-9b47550554bc75b6fb07dd141b5d1c6249aca44f3753f861f551f6814347b953R70)

**Comment by vados-cosmonic** [comment]
This was resolved by #619
